lib/tests/misc.nix

```diff
@@ -831,6 +831,26 @@ runTests {
     };
   };
 
+  testMatchAttrsMatchingExact = {
+    expr = matchAttrs { cpu = { bits = 64; }; } { cpu = { bits = 64; }; };
+    expected = true;
+  };
```

Comment:
Nice, all of these would be great to have as examples in the function docs!

Reply:
While true, I want to keep the scope of this PR slim.